cartridges/int_alma/cartridge/scripts/helpers/almaCheckoutHelper.js

```diff
@@ -159,6 +165,44 @@ function isFragmentActivated() {
     return Site.getCurrent().getCustomPreferenceValue('ALMA_Fragment_Payment');
 }
 
+/**
+ * Return true if plan is activated
+ * @param {Object} paymentMethod payment method
+ * @param {Object} plan plan
+ * @returns {boolean} payment method ID
+ */
+function planIsActivated(paymentMethod, plan) {
+    var almaActivated = paymentMethod.getCustom().almaActivated.split(' | ');
+    var isActivated = false;
+
+    almaActivated.forEach(function (element) {
+        if (element.includes(plan.installments_count || plan.deferred_days)) {
+            isActivated = true;
+        }
+    });
+
+    return isActivated;
+}
+
+/**
+ * Add payment method to plans
+ * @returns {string} method payment
+ * @param {Object} plan plan
+ */
+function getPlanPaymentMethodID(plan) {
+    if (plan.installments_count < 5 && planIsActivated(PaymentMgr.getPaymentMethod(ALMA_PNX_ID), plan)) {
+        paymentMethodId = ALMA_PNX_ID;
+    }
+    if (plan.installments_count >= 5 && planIsActivated(PaymentMgr.getPaymentMethod(ALMA_CREDIT_ID), plan)) {
+        paymentMethodId = ALMA_CREDIT_ID;
+    }
+    if (plan.deferred_days > 0 && planIsActivated(PaymentMgr.getPaymentMethod(ALMA_DEFERRED_ID), plan)) {
```

Comment:
what if I get a plan with both deferred and paylater ?
It shouldn't happend, for now, so there's nothing to do yet, but let's keep that in mind for the future :)

Reply:
I keep it in mind and I will see when I have this case.